{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Week 5 Assignment\n",
    "\n",
    "Welcome to week 5 of informatics II. By now you learned how to format a string. This week assignment is to use stringformatting in order to print out Phredscore and Acurracy alligned. You will aply the template use functions and string formatting. An example is given below\n",
    "\n",
    "    Quality score: #>>1A1B3B1\n",
    "    Phredscore   :    2    |   29   |   29   |   16   |   32   |   16   |   33   |   18   |   33   |   16   |\n",
    "    Accuracy     :  36.90% | 99.87% | 99.87% | 97.49% | 99.94% | 97.49% | 99.95% | 98.42% | 99.95% | 97.49% |\n",
    "\n",
    "\n",
    "Learning outcomes\n",
    "\n",
    "- develop a program from problem to script\n",
    "- apply string formatting to generate a outcome\n",
    "\n",
    "To fulfill the assingment you \n",
    "\n",
    "1. read the [background](#0) to understand the purpose of the to be coded script\n",
    "2. [design the script](#1) using a flow chart\n",
    "3. translate the pieced of the flow chart into pseudo code\n",
    "4. code the script\n",
    "5. [test](#2) the script\n",
    "\n",
    "When you upload your code to blackboard you will be reviewed on the following aspects\n",
    "\n",
    "- Is the code according the template\n",
    "- Is your code according coding style\n",
    "- Is the output of your program is correct\n",
    "- Is the printed layout nicely alligned\n",
    "- The script does not generate errors \n",
    "\n",
    "\n",
    "<a name='0'></a>\n",
    "## Casus Background"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A great portion of the data produced in computational biology is from so called *Next-Generation Sequencers*. These machines read DNA or RNA material and write these sequences to a file (you will learn more about the machines and techniques in the module *Theory of Bioinformatics*). \n",
    "\n",
    "Below you will find two (shortened) reads coming from an Illumina next-generation sequencer. Keep in mind that a typical run on a NGS machine has *millions* of these short sequences!\n",
    "\n",
    "**Read 1**:\n",
    "\n",
    "<font color=\"red\">@M01785:20:000000000-A3F6F:1:1101:16810:1655 1:N:0:2</font>\n",
    "<font color=\"blue\">NTCATGTACGGTCAGGATGGACGCACTCAACATTTTCAAGTTATTACTCCTTCAACTCAAAACTCCAGAAGTACACTAAATCATATATGTTGTTTTCT</font> ...<br />\n",
    "`+` <font color=\"green\"><br />`#`>>1A1B3B11BAEFFBECA0B000EEGFCGBFGGHH2DEGGHGFFFGFFHHFGBGEFFFFFGGEGBF1BCFFE2BGFHBGHGHFF2FFFGHHHHHH</font> ...\n",
    " \n",
    " **Read 2**:\n",
    "\n",
    "<font color=\"red\">@M01785:20:000000000-A3F6F:1:1101:12839:1664 1:N:0:2</font>\n",
    "<font color=\"blue\">\n",
    "TATATCTATGTCATTTTTTTCTCAATAATACTAAGAGAAAGAAGGCAACTCAAGGATCCTATTAATCCTTTAGAATTTCTACTTAAATCTCACATCCATTA</font> ...<br />\n",
    "`+` <font color=\"green\"><br />1>1AFFFD3DDDGGGGGGGGGHF3FDFGFHHFB1110FF10000FGGGHHDC110FEGGBGHFFHFHHHHGBFHHHHHHHHHGHHFFHHHHHHH</font> ...\n",
    " \n",
    "Each read constists of:\n",
    "<ul>\n",
    "<li><font color=\"red\">A first line describing the machine it was run on, the chip identifier and the actual coordinates from the chip where the base was read</font>\n",
    "<li><font color=\"blue\">The actual read sequence</font>\n",
    "<li>The plus sign\n",
    "<li><font color=\"green\">The predicted read base quality ASCII value</font>\n",
    "</ul>\n",
    "\n",
    "The difference with the widely known [FASTA](https://en.wikipedia.org/wiki/FASTA_format) format is mainly the addition of the **Quality** line (green). So given these quality-characters, how do we determine if the above sequences are *good* sequences? Each character in the quality line corresponds with a numerical quality *[score](https://en.wikipedia.org/wiki/Phred_quality_score)*, which can be looked up in so called <a href = \"https://simple.wikipedia.org/wiki/ASCII\">ASCII tables</a>. \n",
    "\n",
    "Lets take a closer look at the first sequence from above. The first 10 bases are: `NTCATGTACG` and the quality scores for these bases are: `#>>1A1B3B1`. We can now look up these quality characters in the ASCII table.\n",
    "<br />\n",
    "If we look for example at the first character `#`, we find the value **35** in the ASCII table. For *Illumina* reads we have to subtract 33 from this value. So we end up with: 35 - 33 = 2. So the score for the first base N is 2.</br>\n",
    "This score is called the *Phred* score. Lets also look at the Phred score for the second base T which has the ASCII character `>`. The `>` character translates to the value **62**. Again subtract 33 from this value to calculate the Phred score. 62 - 33 = **29**.\n",
    "\n",
    "What does the quality score really mean? The score indicates the *probability that the base call is erroneous*. The quality (Phred) score Q is logarithmically related to the probability of an incorrect base call:\n",
    "\n",
    "$ Q = log10P $ or $ P = 10^{(-Q/10)} $. To calculate the probability that our first base was incorrectly called, we can calculate it like this: $$ Q = 2 \\rightarrow P = 10^{(-2/10)} \\rightarrow P = 0,63 $$ \n",
    "\n",
    "which equates to **63%**. We can also look for the probability that the first base was *correct*, then we have to subtract that number of 1. So the probability that the first base was correct is: `1 - 0,63 = 0,37` or 37%. The probability the second base was correctly called is:  $ 1 - (10^{-29/10}) = 0,9987 $ or 99,87% accuracy. In general we can say that any *Phred-score* above 30 is acceptable (a 99,9% accuracy) which both the first two bases fail to get. \n",
    "\n",
    "## Assignment 1\n",
    "\n",
    "Complete the table below.\n",
    "<table style=\"width:100%\">\n",
    "  <tr>\n",
    "   <td>Base</td>\n",
    "     <td>N</td>\n",
    "     <td>T</td>\n",
    "     <td>C</td>\n",
    "     <td>A</td>\n",
    "     <td>T</td>\n",
    "     <td>G</td>\n",
    "     <td>T</td>\n",
    "     <td>A</td>\n",
    "     <td>C</td>\n",
    "     <td>G</td>\n",
    "   </tr>\n",
    "     <tr>\n",
    "   <td>Quality char</td>\n",
    "     <td>#</td>\n",
    "     <td>></td>\n",
    "     <td>></td>\n",
    "     <td>1</td>\n",
    "     <td>A</td>\n",
    "     <td>1</td>\n",
    "     <td>B</td>\n",
    "     <td>3</td>\n",
    "     <td>B</td>\n",
    "     <td>1</td>\n",
    "   </tr>\n",
    "     <tr>\n",
    "   <td>Numerical score (ASCII value - 33)</td>\n",
    "     <td>2</td>\n",
    "     <td>29</td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "   </tr>\n",
    "       <tr>\n",
    "   <td>Base call accuracy (1 - P)</td>\n",
    "     <td>37%</td>\n",
    "     <td>99,87%</td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "     <td></td>\n",
    "    </tr>\n",
    " </table> "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a name='1'></a>\n",
    "## Assignment 2\n",
    "We now will program the procedure. The program should read the fastq file <a href=\"https://bioinf.nl/~fennaf/informatica2/reads.fq\">reads.fq</a>, read the line with the quality ASCII values and calculate the Phredscore and quality base. The Phredscore and the Accuracy should be printed nicely alligned. You can test your program with the test string #>>1A1B3B1. \n",
    "\n",
    "To fulfill the assignment you can conduct the following steps:\n",
    "\n",
    "1. type the template\n",
    "2. determine the steps of the workflow (read file, read qualityline, for each character calculate phredscore and quality score, print results)\n",
    "3. write for each step a function and determine arguments and return (what needs to go in and out)\n",
    "4. for each step test the outcome \n",
    "\n",
    "<a name='2'></a>\n",
    "## Test your outcome\n",
    "The final outcome of your script using the test string #>>1A1B3B1 should be:\n",
    "\n",
    "    Quality score: #>>1A1B3B1\n",
    "    Phredscore   :    2    |   29   |   29   |   16   |   32   |   16   |   33   |   18   |   33   |   16   |\n",
    "    Accuracy     :  36.90% | 99.87% | 99.87% | 97.49% | 99.94% | 97.49% | 99.95% | 98.42% | 99.95% | 97.49% |\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<details>    \n",
    "<summary>\n",
    "    <font size=\"3\" color=\"darkgreen\"><b>Hints</b></font>\n",
    "</summary>\n",
    "\n",
    "You can add spaces (padding) to a string in Python or format a strings by defining the format between the bracket. N is total length\n",
    "\n",
    "    0 Padding\t{variable:0N}\n",
    "    Decimal Places\t{variable:.Nf}\n",
    "    Date Formatting\t{date : Directive}\n",
    "    Space Padding\t{variable:N}\n",
    "    Justified string\t{variable:>N}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check your code style\n",
    "\n",
    "Before submitting your code (for instance qscore.py) you can check your code using a lintr by typing in the commandline\n",
    "\n",
    "    pylint qscore.py\n",
    "    \n",
    "Once your code complies with the coding style and the assignment requirements you can submit your solution in blackboard"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}